frame/dex-router/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+//! # DEX Router Pallet
+//!
+//! Is used to add route to DEX for given asset_id's pair.
+
+#![cfg_attr(not(test), warn(clippy::disallowed_method, clippy::indexing_slicing))] // allow in tests
+#![warn(clippy::unseparated_literal_suffix, clippy::disallowed_type)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use codec::FullCodec;
+use composable_traits::{
+		currency::AssetIdLike,
+	    dex::{DexRoute, DexRouter},
+	};
+	use frame_support::pallet_prelude::*;
+
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type AssetId: AssetIdLike + Decode + Clone + core::fmt::Debug + Default;
+        type RouteId: FullCodec + Copy + Eq + PartialEq + core::fmt::Debug + TypeInfo;
+        type PoolId: FullCodec + Copy + Eq + PartialEq + core::fmt::Debug + TypeInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+    pub type DexRoutes<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        T::AssetId,
+        Blake2_128Concat,
+        T::AssetId,
+        DexRoute<T::PoolId, T::RouteId>,
+        OptionQuery,
+    >;
+
+	#[pallet::error]
+	pub enum Error<T> {
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+        RouteAdded { x_asset_id: T::AssetId, y_asset_id: T::AssetId, route : DexRoute<T::PoolId, T::RouteId> },
+        RouteDeleted { x_asset_id: T::AssetId, y_asset_id: T::AssetId, route : DexRoute<T::PoolId, T::RouteId> },
+        RouteUpdated { x_asset_id: T::AssetId, y_asset_id: T::AssetId, route : DexRoute<T::PoolId, T::RouteId> },
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+	}
+
+	impl<T: Config> Pallet<T> {
+	}
+
+    impl<T: Config> DexRouter<T::AssetId, T::PoolId, T::RouteId> for Pallet<T> {
+        fn update(asset_pair : (T::AssetId, T::AssetId), route : Option<DexRoute<T::PoolId, T::RouteId>>) {
+            match route {
+                Some(r) => {
+                    let mut update = false;
+                    if DexRoutes::<T>::contains_key(asset_pair.0, asset_pair.1) {
```

Comment:
would you make TODO to use CurrenchPair?

Reply:
once your PR will be merged I will use that.